esp-hal/src/dma/mod.rs

```diff
@@ -573,6 +588,32 @@ macro_rules! dma_circular_descriptors_chunk_size {
     };
 }
 
+/// Convenience macro to create a DmaTxBuf from buffer size. The buffer and
+/// descriptors are statically allocated and used to create the `DmaTxBuf`.
+///
+/// ## Usage
+/// ```rust,no_run
+#[doc = crate::before_snippet!()]
+/// use esp_hal::dma_tx_buffer;
+/// use esp_hal::dma::DmaBufBlkSize;
+///
+/// let tx_buf =
+///     dma_tx_buffer!(32000);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! dma_tx_buffer {
+    ($tx_size:expr) => {{
+        const TX_DESCRIPTOR_LEN: usize =
+            $crate::dma::DmaTxBuf::compute_descriptor_count($tx_size, None);
+        static mut TX_BUFFER: [u8; $tx_size] = [0u8; $tx_size];
+        static mut TX_DESCRIPTORS: [$crate::dma::DmaDescriptor; TX_DESCRIPTOR_LEN] =
+            [$crate::dma::DmaDescriptor::EMPTY; TX_DESCRIPTOR_LEN];
+        let (tx_buffer, tx_descriptors) = unsafe { (&mut TX_BUFFER, &mut TX_DESCRIPTORS) };
+        $crate::dma::DmaTxBuf::new(tx_descriptors, tx_buffer)
```

Comment:
I'd recommend defining a `dma_buffer` macro that returns the (buffer, descriptor) tuple, and reimplement this and the other macros in terms of that macro.

Reply:
Maybe that can be a seperate PR. As TX and TX alignments differ when using PSRAM.

Reply:
maybe we should drop it from this PR and revisit when we could also handle `dma_rx_buffer`...  Besides - I don't have anything that even uses it at the moment.  We could convert the write only spi tests and examples maybe?  But
then we have two styles of building the DmaTxBuf when compared with the read/write tests and examples.  (and it would be really ugly to convert the TX part of those)

Reply:
We should definitely have a separate PR that just revamps all the DMA macros to do the stuff we've been talking about.